types/protractor-http-mock/index.d.ts

```diff
@@ -108,31 +104,37 @@ declare namespace mock {
     }
 
     /**
-     * Plugin for custom matching logic.
+     * Plugin for custom matching logic with 1 generic type.
      */
-    interface Plugin {
+    interface Plugin1<T1> {
         /**
          * Match function.
          * Return a truthy value to indicate successfull match.
          *
          * @param mockRequest The mock to compare request with.
          * @param requestConfig The request object to compare mock with.
          */
-        match<T1, T2>(mockRequest: requests.AllRequests<T1, T2>, requestConfig: requests.AllRequests<T1, T2>): boolean;
```

Comment:
I'm overloading the method to reflect the fact different types of requests are not compared.
Prior to calling a plug-in types of both sides are compared in `matchMethod()`. Only after that `matchByPlugins()` is invoked.

https://github.com/atecarlos/protractor-http-mock/blob/77df9c34a6cb837856fdb23e9cfb580a51ea4aec/lib/httpMock.js#L187